SalesforceFoundation/HEDAP
# fix the duplicate field issue

# Critical Changes

# Changes
- Fixed an error that prevented you from mass updating the record type on Accounts. This duplicate field error occurred whenever the original record type value didn't have a primary affiliation mapping and the new record type value you were updating to did.

- Fixed an issue that incorrectly populated the primary affiliation field on Contact when mass updating the record type on Accounts. The issue occurred whenever the updates involved multiple record types and the original record type value didn't have a primary affiliation mapping and the new record type value you were updating to did.

# Issues Closed
Fixes #518 
Fixes #524 
# New Metadata

# Deleted Metadata

# Testing Notes
Steps to reproduce the issue #518  :
1. Create multiple Accounts with Administrative record type.
2. Update those Accounts record type from Administrative to other record type, such as  "Academic Program".  Please note that you need to do the update in bulk instead of one by one. Also, the record type that you change to must have a mapping in "Affl Settings". 
3. System will display the error "hed.TDTM_Account: execution of AfterUpdate
caused by: System.QueryException: duplicate field selected: Primary_Academic_Program__c" 

Verify the fix:
You should not get any error when you do the steps above. 

Steps to reproduce the issue #524 :
1. Create two Contacts: A, B
2. Create two Accounts: Math, Football. Set both record type to Admin first
3. Create two Affiliations: A - Math, B-Football. Currently, the primary fields on A and B should be empty
4. Update Accounts in one operation: Math - Academic Program, Football - sports organization
5. System should update Primary Academic Program of A and the Primary sports org of B. However, you will see both Primary Academic Program, Primary sports org of A becomes Math (similar thing happen to B). Which is not correct.
